utils/filesystem/filepath.go

```diff
@@ -34,3 +40,97 @@ func FileTreeDepth(fs FS, root, filePath string) (depth int64, err error) {
 func EndsWithPathSeparator(fs FS, filePath string) bool {
 	return strings.HasSuffix(filePath, "/") || strings.HasSuffix(filePath, string(fs.PathSeparator()))
 }
+
+// NewPathValidationRule returns a validation rule to use in configuration.
+// The rule checks whether a string is a valid not empty path.
+// `when` describes whether the rule is enforced or not
+func NewPathValidationRule(filesystem FS, when bool) validation.Rule {
+	return &pathValidationRule{condition: when, filesystem: filesystem}
+}
+
+// NewOSPathValidationRule returns a validation rule to use in configuration.
+// The rule checks whether a string is a valid path for the Operating System's filesystem.
+// `when` describes whether the rule is enforced or not
+func NewOSPathValidationRule(when bool) validation.Rule {
+	return NewPathValidationRule(GetGlobalFileSystem(), when)
+}
+
+type pathValidationRule struct {
+	condition  bool
+	filesystem FS
+}
+
+func (r *pathValidationRule) Validate(value interface{}) error {
+	err := validation.Required.When(r.condition).Validate(value)
+	if err != nil {
+		return fmt.Errorf("%w: path [%v] is required: %v", commonerrors.ErrUndefined, value, err.Error())
+	}
+	if !r.condition {
+		return nil
+	}
+	pathString, err := validation.EnsureString(value)
+	if err != nil {
+		return fmt.Errorf("%w: path [%v] must be a string: %v", commonerrors.ErrInvalid, value, err.Error())
+	}
+	pathString = strings.TrimSpace(pathString)
+	if r.filesystem.Exists(pathString) {
+		return nil
+	}
+
+	// Inspired from https://github.com/go-playground/validator/blob/84254aeb5a59e615ec0b66ab53b988bc0677f55e/baked_in.go#L1604 and https://stackoverflow.com/questions/35231846/golang-check-if-string-is-valid-path
+	if pathString == "" {
+		return fmt.Errorf("%w: the path [%v] is empty", commonerrors.ErrUndefined, value)
+	}
+	if _, err := r.filesystem.Stat(pathString); err != nil {
+		switch t := err.(type) {
+		case *fs.PathError:
+			if t.Err == syscall.EINVAL {
+				return fmt.Errorf("%w: the path [%v] has invalid characters: %v", commonerrors.ErrInvalid, value, err.Error())
+			}
+		}
+	}
+	if strings.Contains(pathString, "\n") {
```

Comment:
I think anything checking the validity of the path should come before the stat and the filesystem.Exists